README.md

```diff
@@ -61,6 +59,190 @@ yarn start
 
 Finally, we can navigate a browser window to `localhost:8080` to view generated data.
 
+### Running with Docker
+
+You can build and run sourcecred in a container to avoid installing dependencies on your host. First, build the container:
+
+```bash
+$ docker build -t sourcecred/sourcecred .
+```
+
+If you want to build and customize the `SOURCECRED_DIRECTORY`, you can set that as a `--build-arg`:
+
+```bash
+$ docker build --build-arg SOURCECRED_DEFAULT_DIRECTORY=/tmp/data -t sourcecred/sourcecred .
+```
+
+Your options for running the container including the following commands. 
+Examples will be shown for each.
+
+ - **dev-preview**: offers a shortcut for loading sourcecred and then starting a dev server. This is likely the option you'll choose if you want to provide a respository or an organization and preview results a web interface.
+ - **dev-server**: exposes several webpack operations without the initial load. This takes no arguments.
+ - **build**: simply provides the build command to yarn, followed by any argumnents that you provide.
+ - **(anything else)**: will be passed on to sourcecred.js
+
+#### Development Preview 
+
+To run the development preview, you will still need to export a GitHub token, and then provide it to the container when you run it. 
+Notice that we are also binding port 8080 so we can view the web interface that will be opened up.  
+The only argument needed is a command to load the GitHub repository to generate the sourcecred for:
+
+```bash
+REPOSITORY=sfosc/sfosc
+$ SOURCECRED_GITHUB_TOKEN="xxxxxxxxxxxxxxxxx" \
+    docker run -d --name sourcecred --rm --env SOURCECRED_GITHUB_TOKEN -p 8080:8080 sourcecred/sourcecred dev-preview "${REPOSITORY}"
+```
+
+You can also specify an entire organization:
+
+```bash
+ORGANIZATION=@sfosc
+$ SOURCECRED_GITHUB_TOKEN="xxxxxxxxxxxxxxxxx" \
+    docker run -d --name sourcecred --rm --env SOURCECRED_GITHUB_TOKEN -p 8080:8080 sourcecred/sourcecred dev-preview "${ORGANIZATION}"
+```
+
+If you want to bind the data folder to the host, you can do that too. 
+In the example below, we have a folder "data" in the present working directory that we bind to "/data" in the container, the default `SOURCECRED_DIRECTORY`.
+
+```bash
+$ SOURCECRED_GITHUB_TOKEN="xxxxxxxxxxxxxxxxx" \
+    docker run -d --name sourcecred --rm --env SOURCECRED_GITHUB_TOKEN -p 8080:8080 -v $PWD/data:/data sourcecred/sourcecred load "${REPOSITORY}"
+```
+
+We are running in detached mode (-d) so it's easier to remove the container after. 
+After running the command, you can inspect it's progress like this:
+
+```bash
+$ docker logs sourcecred
+  GO   load-sfosc/sfosc
+  GO   github/sfosc/sfosc
+ DONE  github/sfosc/sfosc: 25s
+  GO   compute-cred
+ DONE  compute-cred: 1s
+ DONE  load-sfosc/sfosc: 26s
+...
+```
+
+It will take about 30 seconds to do the initial build, and when the web server is running you'll see this at the end:
```

Comment:
The load command does not build or run a server. So here we can show the dev-server command.

```
docker run -d --name sourcecred --rm -p 8080:8080 -v $PWD/data:/data sourcecred/sourcecred dev-server
```

Reply:
oup and you just beat me to my last comment :)